### PR TITLE
[Android] Add response headers to XHR response

### PR DESCRIPTION
**Pull request by satya164**

### CODE REVIEW

File: ReactAndroid/src/main/java/com/facebook/react/modules/network/NetworkingModule.java [hunk 1]
```diff
@@ -189,7 +190,15 @@ public void onResponse(Response response) throws IOException {
               callback.invoke(0, null, e.getMessage());
               return;
             }
-            callback.invoke(response.code(), null, responseBody);
+
+            WritableMap responseHeaders = Arguments.createMap();
+            Headers headers = response.headers();
+
+            for (int i = 0; i < headers.size(); i++) {
+              responseHeaders.putString(headers.name(i), headers.value(i));
```
Comment (ide) [inline ReactAndroid/src/main/java/com/facebook/react/modules/network/NetworkingModule.java]: We'll lose data if there are multiple values for the same header. okhttp has a method to get a list of values for a single header, for example.

**Comment by satya164** [comment]
@ide Ok. I'll check that out and update the pull request accordingly.

File: ReactAndroid/src/main/java/com/facebook/react/modules/network/NetworkingModule.java (hunk 1 above)
Comment (qbig) [inline ReactAndroid/src/main/java/com/facebook/react/modules/network/NetworkingModule.java]: change 
```java
responseHeaders.putString(headers.name(i), headers.value(i));
```
to
```java
if (responseHeaders.hasKey(headers.name(i)) {
  responseHeaders.putString(
    headers.name(i),
    responseHeaders.getString(headers.name(i) + headers.value(i)
  ) 
} else {
  responseHeaders.putString(headers.name(i), headers.value(i));}
}
```
would do ? @ide @satya164

**Comment by nucleartux** [comment]
any updates?

**Comment by satya164** [comment]
Closing this since @nucleartux has a better pull request.

**Pull request closed without merging**